tests/QuikGraph.Tests/Algorithms/Contracts/VertexColorizerContract.cs

```diff
@@ -54,129 +62,126 @@ public void ExceptionThrown_WhenAlgorithmHasNotYetBeenComputed()
                 SingleVerticesInGraph = new int[0],
-                AccessibleVerticesFromRoot = new[] {2},
+                AccessibleVerticesFromRoot = new[] { 2 },
                 Root = 1,
                 DoComputation = false
             };
 
-            var algorithm = CreateAlgorithmAndMaybeDoComputation(scenario);
+            IVertexColorizerAlgorithm<int> algorithm = CreateAlgorithmAndMaybeDoComputation(scenario);
 
-            var expectedExceptionType = GetExpectedExceptionType();
+            Type expectedExceptionType = GetExpectedExceptionType();
             Assert.Throws(expectedExceptionType, () => algorithm.GetVertexColor(2));
 
+            #region Local function
+
             Type GetExpectedExceptionType()
             {
-                switch (_classUnderTest)
+                switch (_testedAlgorithm)
                 {
-                    case var _ when _classUnderTest == typeof(AStarShortestPathAlgorithm<,>):
-                    case var _ when _classUnderTest == typeof(BellmanFordShortestPathAlgorithm<,>):
-                    case var _ when _classUnderTest == typeof(DagShortestPathAlgorithm<,>):
-                    case var _ when _classUnderTest == typeof(DijkstraShortestPathAlgorithm<,>):
-                    case var _ when _classUnderTest == typeof(UndirectedDijkstraShortestPathAlgorithm<,>):
+                    case var _ when _testedAlgorithm == typeof(AStarShortestPathAlgorithm<,>):
+                    case var _ when _testedAlgorithm == typeof(BellmanFordShortestPathAlgorithm<,>):
+                    case var _ when _testedAlgorithm == typeof(DagShortestPathAlgorithm<,>):
+                    case var _ when _testedAlgorithm == typeof(DijkstraShortestPathAlgorithm<,>):
+                    case var _ when _testedAlgorithm == typeof(UndirectedDijkstraShortestPathAlgorithm<,>):
                         return typeof(NullReferenceException);
-
-                    case var _ when _classUnderTest == typeof(BidirectionalDepthFirstSearchAlgorithm<,>):
-                    case var _ when _classUnderTest == typeof(BreadthFirstSearchAlgorithm<,>):
-                    case var _ when _classUnderTest == typeof(CyclePoppingRandomTreeAlgorithm<,>):
-                    case var _ when _classUnderTest == typeof(DepthFirstSearchAlgorithm<,>):
-                    case var _ when _classUnderTest == typeof(EdmondsKarpMaximumFlowAlgorithm<,>):
-                    case var _ when _classUnderTest == typeof(TSP<,,>):
-                    case var _ when _classUnderTest == typeof(UndirectedBreadthFirstSearchAlgorithm<,>):
-                    case var _ when _classUnderTest == typeof(UndirectedDepthFirstSearchAlgorithm<,>):
-                        return typeof(VertexNotFoundException);
-
-                    default:
-                        throw new AssertionException($"No test constructor known for {_classUnderTest}");
                 }
+
+                return typeof(VertexNotFoundException);
```

Comment:
This is much more readable which is nice. However, there is a danger that if a new algorithm implementation is added and the switch statement isn't updated we won't get a warning that something is wrong in case it should throw a NullReferenceException but throws a VertexNotFoundException. It is probably not a big issue though, so keeping it like this is also fine for me.